### PR TITLE
implement userAnswer

### PR DESCRIPTION
**Pull request by vlad12345123**

### CODE REVIEW

File: exalearn-api/ExaLearn.WebApi/Controllers/UserAnswerController.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿using ExaLearn.Bl.DTO;
+using ExaLearn.Bl.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+
+namespace ExaLearn.WebApi.Controllers
+{
+    [Route("api/userAnswer")]
```
Comment (romanstat) [inline exalearn-api/ExaLearn.WebApi/Controllers/UserAnswerController.cs]: s

File: exalearn-api/ExaLearn.WebApi/Controllers/UserAnswerController.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+        private readonly IUserAnswerService _userAnswerService;
+
+        public UserAnswerController(IUserAnswerService userAnswerService)
+        {
+            _userAnswerService = userAnswerService;
+        }
+
+        [HttpPost("add")]
+        public IActionResult Add(List<UserAnswerDTO> userAnswers)
```
Comment (romanstat) [inline exalearn-api/ExaLearn.WebApi/Controllers/UserAnswerController.cs]: async

File: exalearn-api/ExaLearn.Dal/Repositories/UserAnswerRepository.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace ExaLearn.Dal.Repositories
+{
+    public class UserAnswerRepository : GenericRepository<UserAnswer>, IUserAnswerRepository
+    {
+        public UserAnswerRepository(ExaLearnDbContext appDbContext) : base(appDbContext)
+        {
+        }
+
+        public async Task<List<UserAnswer>> AddUserAnswers(List<UserAnswer> userAnswers)
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Dal/Repositories/UserAnswerRepository.cs]: naming

File: exalearn-api/ExaLearn.Bl/Services/UserAnswerService.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+            _userAnswerRepository = userAnswerRepository;
+            _mapper = mapper;
+        }
+
+        public async Task<List<UserAnswerDTO>> AddUserAnswers(List<UserAnswerDTO> userAnswersDTO)
+        {
+            var userAnswer = _mapper.Map<List<UserAnswer>>(userAnswersDTO);
+            userAnswer =  await _userAnswerRepository.AddUserAnswers(userAnswer);
+            return _mapper.Map<List<UserAnswerDTO>>(userAnswer);
+        }
```
Comment (artemgael96) [inline exalearn-api/ExaLearn.Bl/Services/UserAnswerService.cs:27]: add test

File: exalearn-api/ExaLearn.Bl/Services/QuestionService.cs [hunk 5]
```diff
@@ -48,5 +50,12 @@ public async Task<TopicQuestionDTO> CreateTopicQuestionAsync(TopicQuestionDTO to
             var question = await _questionRepository.CreateAsync(_mapper.Map<Question>(topicQuestionDTO));
             return _mapper.Map<TopicQuestionDTO>(question);
         }
+
+        public async Task<PassedTestDTO> CreatePassedTestAsync(PassedTestDTO passedTestDTO)
+        {
+            var passedTest = _mapper.Map<PassedTest>(passedTestDTO);
+            passedTest = await _passedTestRepository.CreateAsync(passedTest);
+            return _mapper.Map<PassedTestDTO>(passedTest);
+        }
```
Comment (artemgael96) [inline exalearn-api/ExaLearn.Bl/Services/QuestionService.cs:59]: add test

File: exalearn-api/ExaLearn.Dal/Entities/AssignTest.cs [hunk 6]
```diff
@@ -9,7 +9,7 @@ public class AssignTest
 
         public int UserId { get; set; }
 
-        public User User { get; set; }
+        public User User { get; set; } //i think will be is many (we can assign test on many users). however, roma, your approval is needed
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Dal/Entities/AssignTest.cs]: remove comment

**Review by romanstat — APPROVED** [review]

**Review by artemgael96 — APPROVED** [review]

**Pull request merged**